src/encoding.rs

```diff
@@ -34,12 +35,13 @@ impl TryFrom<pkcs8::PrivateKeyInfo<'_>> for RsaPrivateKey {
         if pkcs1_key.version() != pkcs1::Version::TwoPrime {
             return Err(pkcs1::Error::Version.into());
         }
-
-        let n = BigUint::from_bytes_be(pkcs1_key.modulus.as_bytes());
-        let e = BigUint::from_bytes_be(pkcs1_key.public_exponent.as_bytes());
-        let d = BigUint::from_bytes_be(pkcs1_key.private_exponent.as_bytes());
-        let prime1 = BigUint::from_bytes_be(pkcs1_key.prime1.as_bytes());
-        let prime2 = BigUint::from_bytes_be(pkcs1_key.prime2.as_bytes());
+        let bits = 512; // TODO: read from data
```

Comment:
You should be able to use `pkcs1_key.modulus.as_bytes().len() * 8` for this, I think?